core/src/main/java/org/apache/iceberg/rest/auth/OAuth2Util.java

```diff
@@ -302,8 +318,16 @@ public static OAuthTokenResponse tokenResponseFromJson(JsonNode json) {
     return builder.build();
   }
 
+  public static boolean tokenExpired(Optional<JWT> jwt) {
+    return jwt.isPresent()
+        && (jwt.get()
+            .expiredAt(System.currentTimeMillis() + AuthSession.MAX_REFRESH_WINDOW_MILLIS));
```

Comment:
I think that checking whether a token is expired should check the current time, not whether it is expected to expire in 5 minutes.

Reply:
makes sense, updated